### ISSUE TITLE
Worldcat Pass-through Failures

### ISSUE THREAD

**Issue by sarahjeansweeney** [NEU-Libraries/scholar-onesearch#124]
From case 00039388: We have noticed that the WorldCat pass-through link ("Request options: Request book from another library") in the Location and Status (Get It!) tab fails intermittently. When the pass-through is successful the OCLC number is correctly included in the link (http://northeastern.worldcat.org/oclc/852257484) and a new page opens in WorldCat Local (http://northeastern.worldcat.org/title/world-war-z/oclc/852257484). When the pass-through fails the OCLC number is missing from the link (http://northeastern.worldcat.org/oclc/) and a new page opens in WorldCat Local with this message: "The page you tried was not found. You may have used an outdated link or may have typed the address (URL) incorrectly.", even when the item does exist in WorldCat Local. We would like this pass-through to work consistently for all physical items with an OCLC number, not just for some. 

![untitled](https://f.cloud.github.com/assets/4663073/2284543/7767a888-9fca-11e3-91c1-bf19c6919006.png)

![untitled2](https://f.cloud.github.com/assets/4663073/2284545/7b80be28-9fca-11e3-9e19-f117b6c851b0.png)

**Comment by sarahjeansweeney** [comment]
I have an update on the WorldCat passthrough issue from last Fall:
In Alma's test environment, Stacy and I made some changes to the rules to stop the "Request from another library" link from displaying in onesearch-test when the record does not have an OCLC number (which was causing those pages of blank results), and that appears to have been successful. 
We tried to add another rule to display the "Request..." link when the record does not have an OCLC number, which would search WorldCat using an ISBN number instead. Unfortunately the Open URL ISBN sends all the ISBN information, including extra text like "pbk", which also causes the WorldCat search to fail. The same happens for a title search.
Ernesto discovered a way around these failures, essentially constructing another link template which performs an advanced search based on the title and author's last name. So when a record does not have an OCLC number, the passthrough link behaves similarly to the "Search WorldCat" link at the top of the search page. This is a pretty great solution, especially for the Primo Central records.

Test it out for yourself here http://onesearch-test.neu.edu/ . Here are some examples to try: 
Les misérables (Has OCLC number: 827611665)
Post-pop art (No OCLC number, just an ISBN: 0262700379)
Acting together on the world stage (No OCLC number, no ISBN)
Art : educating with art without ruining it / Gringras, Robbie (TN_rambi000514126 – Primo Central record)
Rights at Risk: Equality in an Age of Terrorism. Report of the Citizens' Commission on Civil Rights / Piche, Dianne M. , Ed. ; Taylor, William L. , Ed. ; Reed, Robin A. , Ed. (TN_ericED463386 – Primo Central record)
Shaping the Reading Field: The Impact of Early Reading Pioneers, Scientific Research, and Progressive Ideas / Israel, Susan E. , Ed. ; Monaghan, E. Jennifer, Ed. (TN_ericED494236 – Primo Central record)
Adolescent Literacy / Ippolito, Jacy (TN_ericED529634 – Primo Central record)

If this is an acceptable solution, we can make these changes in production.

**Comment by sarahjeansweeney** [comment]
I applied the change to production, and it seems to be working just fine. My next step is to tweak the rules in staging to get better results, then apply those tweaks to production.

While updating the rules I noticed that there is a stray "Request" button:
![2014-03-18_1538](https://f.cloud.github.com/assets/4663073/2452744/451ae970-aed5-11e3-9fcd-09a85c61c300.png)

I'm not sure where this comes from or how to remove it. It does not display in staging.
